integration_tests/src/main/python/aqe_test.py

```diff
@@ -121,3 +121,72 @@ def do_join(spark):
         return spark.sql("select a.* from {} a, {} b where a.name=b.name".format(
             resultdf_name, resultdf_name))
     assert_gpu_and_cpu_are_equal_collect(do_join, conf=_adaptive_conf)
+
+
+joins = [
+    'inner',
+    'cross',
+    'left',
+    'left outer',
+    'left semi',
+    'right',
+    'right outer',
+    'full',
+    'full outer',
+    'left anti',
+    'anti'
+]
+
+# see https://github.com/NVIDIA/spark-rapids/issues/7037
+# basically this happens when a GPU broadcast exchange is reused from 
+# one side of a GPU broadcast join to be used on one side of a CPU 
+# broadcast join. The bug currently manifests in Databricks, but could
+# theoretically show up in other Spark distributions
+@ignore_order(local=True)
+@allow_non_gpu('BroadcastNestedLoopJoinExec', 'BroadcastExchangeExec', 'Cast', 'DateSub')
+@pytest.mark.parametrize('join', joins, ids=idfn)
```

Comment:
I'm fine with reducing to a handful of join types, basically removing the left, right and full outer joins and leaving inner (due to Databricks), left semi, and left anti and anti joins (where the broadcast to row seems to be needed in Databricks). The concern I have is that since sometimes the planner/optimizer might decide to do different things in the future based on the join (for instance in Databricks the join was actually converted from inner join to left semi join), the worry is that some other version of this bug will happen with a different join specified in SQL and we would be missing some test coverage for it.  Note that this bug doesn't even manifest in Databricks 9.1 at the moment, and we also don't know when it can manifest in Apache Spark or other distributions (like EMR as described in #6978).